mutmut/node_mutation.py

```diff
@@ -96,6 +96,35 @@ def operator_arg_removal(
             yield node.with_changes(args=[*node.args[:i], *node.args[i + 1 :]])
 
 
+supported_str_methods_swap = [
+         ("lower", "upper"),
+         ("upper", "lower"),
+         ("lstrip", "rstrip"),
+         ("rstrip", "lstrip"),
+         ("find", "rfind"),
+         ("rfind", "find"),
+         ("ljust", "rjust"),
+         ("rjust", "ljust"),
+         ("index", "rindex"),
+         ("rindex", "index"),
+         ("split", "rsplit"),
+         ("rsplit", "split"),
+         ("removeprefix", "removesuffix"),
+         ("removesuffix", "removeprefix"),
+         ("partition", "rpartition"),
+         ("rpartition", "partition")
+     ]
+
+def operator_string_methods_swap(
+     node: cst.Call
+ ) -> Iterable[cst.Call]:
+     """try to swap string method to opposite e.g. a.lower() -> a.upper()"""
+
+     for old_call, new_call in supported_str_methods_swap:
+         if m.matches(node.func, m.Attribute(value=m.DoNotCare(),  attr=m.Name(value=old_call))):
+             yield node.with_changes(func=cst.Attribute(value=node.func.value, attr=cst.Name(value=new_call)))
```

Comment:
I'd prefer to use `with_changes` (or `with_deep_changes`) to only modify the function name and cloning the rest as it was before. This makes sure that whitespace and other stuff belonging to the attribute is copied, and we only change the function name.

```suggestion
            func_name = cst.ensure_type(node.func, cst.Attribute).attr
            yield node.with_deep_changes(func_name, value=new_call)
```

Reply:
Thank you a lot, that my first time with libcst! I definitely agree with your preferences, it's much better

Reply:
You're welcome :) Thank you for your contribution!